### PR TITLE
Feature: parse JUnit-XML with testsuites top-level element

### LINKED ISSUE #27: `<testsuites>` JUnit Root Element
Some tools produce a root `<testsuites>` element for JUnit results. Currently this tag has to be removed before the results can be used here.

`<testsuite>` and `<testsuites>` should both be supported as the root element.

See: https://github.com/testmoapp/junitxml/tree/main
> `<testsuites>`: Usually the root element of a JUnit XML file. Some tools leave out 
the `<testsuites>` element if there is only a single top-level `<testsuite>` element (which
is then used as the root element).

<details><summary>Error Message</summary>
<pre>
javax.xml.bind.UnmarshalException: unexpected element (uri:"", local:"testsuites"). Expected elements are <{}error>,<{}failure>,<{}successInfo>,<{}testcase>,<{}testsuite>
	at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:712)
	at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:232)
	[...]
	at javax.xml.bind.helpers.AbstractUnmarshallerImpl.unmarshal(AbstractUnmarshallerImpl.java:209)
	at de.tum.in.www1.jenkins.notifications.SendTestResultsNotificationPostBuildTask.lambda$extractTestResults$1(SendTestResultsNotificationPostBuildTask.java:121)
[...]
</pre>
</details>

### PR DESCRIPTION
**Pull request by b-fein**
Closes #27.

Note: The first commit has only the Spotless reformatting changes, the second commit contains the actual logic changes.

Todo:

- [x] test on Uni Passau testserver

---

Todo @b-fein: Once this is merged and a new release is published, open a PR in the main Artemis repository to remove the `sed` workarounds from the `pipeline.groovy` Jenkins build plans.

### CODE REVIEW

File: src/main/java/de/tum/in/www1/jenkins/notifications/SendTestResultsNotificationPostBuildTask.java
Comment (b-fein) [inline src/main/java/de/tum/in/www1/jenkins/notifications/SendTestResultsNotificationPostBuildTask.java:1]: Changes in this file are mostly formatting changes due to Spotless. Commits f41c2d143c3192c31db5ab27072d8945893e7e7c and b7b1d794f7ccc6e7c6706ee92a0e54ec7bf071c0 contain the actual changes.

**Review by magaupp — CHANGES_REQUESTED** [review]
Nested `testsuite`s are not handled correctly with multiple top-most `testsuite`s.
The top-most `testsuite`s' names are not prepended to the inner `testsuite`s.

```xml
<testsuites>
    <testsuite name="SuiteA" tests="3">
        <testsuite name="SuiteC" tests="3">
            <testcase name="Test1"/>
            <testcase name="Test2"/>
            <testcase name="Test3"/>
        </testsuite>
    </testsuite>
    <testsuite name="SuiteB" tests="3">
        <testsuite name="SuiteD" tests="3">
            <testcase name="Test1"/>
            <testcase name="Test2"/>
            <testcase name="Test3"/>
        </testsuite>
    </testsuite>
</testsuites>
```

```
Expected :[SuiteA.SuiteC.Test1, SuiteA.SuiteC.Test2, SuiteA.SuiteC.Test3, SuiteB.SuiteD.Test1, SuiteB.SuiteD.Test2, SuiteB.SuiteD.Test3]
Actual   :[SuiteC.Test1, SuiteC.Test2, SuiteC.Test3, SuiteD.Test1, SuiteD.Test2, SuiteD.Test3]
```

**Comment by b-fein** [comment]
@magaupp Good catch! Thanks. I fixed it and added your example as additional test case.

**Review by magaupp — APPROVED** [review]
Code looks good now :+1:

**Pull request merged**